lib/recipe.rb

```diff
@@ -0,0 +1,25 @@
+class Recipe
+  attr_reader :name, :ingredients_required
+  def initialize(name)
+    @name = name
+    @ingredients_required = {}
+  end
+
+  def add_ingredient(ingredient_obj, amount)
+    if @ingredients_required.keys.include?(ingredient_obj)
+      @ingredients_required[ingredient_obj] += amount
+    else
+      @ingredients_required[ingredient_obj] = amount
+    end
+  end
```

Comment:
you should be storing ingredients, this makes ingredients in cookbook easier to manage